app/services/CheckoutService.scala

```diff
@@ -0,0 +1,54 @@
+package services
+
+import java.util.NoSuchElementException
+
+import com.typesafe.scalalogging.LazyLogging
+import model.SubscriptionData
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.Left
+
+sealed class CheckoutException extends RuntimeException
+object GuestUserNotCreated extends CheckoutException
+object InvalidLoginCookie extends CheckoutException
+object SalesforceUserNotCreated extends CheckoutException
+
+class CheckoutService(identityService: IdentityService, salesforceService: SalesforceService) extends LazyLogging {
+
+  def processSubscription(subscriptionData: SubscriptionData, scGuUCookie: Option[String] = None): Future[Either[CheckoutException, String]] = {
+
+    val preserveFailureReason = (ex: CheckoutException) => Future.successful(Left(ex))
+
+    val lookupUser: Future[Either[CheckoutException, IdUser]] =
+      scGuUCookie.map(
+        identityService.userLookupByScGuU(_)
+          .map(_.toRight(InvalidLoginCookie)))
+        .orElse(
+          Some(identityService.userLookupByEmail(subscriptionData.personalData.email)
+            .filter(_.isDefined)
+            .recoverWith { case _: NoSuchElementException => identityService.registerGuest(subscriptionData.personalData) }
```

Comment:
I kept the case on the same line for brevity since there is only one case. I can put the `registerGuest` call on the next line for readability. Will that work in your opinion?